controllers/spacerequest/spacerequest_controller.go

```diff
@@ -104,18 +104,19 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, err
 	}
 
-	subSpace, createdOrUpdated, err := r.ensureSpace(ctx, memberClusterWithSpaceRequest, spaceRequest)
+	subSpace, tier, createdOrUpdated, err := r.ensureSpace(ctx, memberClusterWithSpaceRequest, spaceRequest)
 	// if there was an error or if subSpace was just created or updated,
 	// let's just return.
 	if err != nil || createdOrUpdated {
 		return ctrl.Result{}, err
 	}
 
-	// ensure there is a secret that provides admin access to each provisioned namespaces of the subSpace
-	if err := r.ensureSecretForProvisionedNamespaces(ctx, memberClusterWithSpaceRequest, spaceRequest, subSpace); err != nil {
-		return reconcile.Result{}, r.setStatusFailedToCreateSubSpace(ctx, memberClusterWithSpaceRequest, spaceRequest, err)
+	if tier.Spec.SpaceRequestConfig != nil {
+		// ensure there is a secret that provides admin access to each provisioned namespaces of the subSpace
+		if err := r.ensureSecretForProvisionedNamespaces(ctx, memberClusterWithSpaceRequest, spaceRequest, subSpace, tier.Spec.SpaceRequestConfig.ServiceAccountName); err != nil {
+			return reconcile.Result{}, r.setStatusFailedToCreateSubSpace(ctx, memberClusterWithSpaceRequest, spaceRequest, err)
+		}
```

Comment:
Unless I've missed it, I don't see if we check also that `tier.Spec.SpaceRequestConfig.ServiceAccountName` is not empty. It can be done here or inside `ensureSecretForProvisionedNamespaces`. 

Probably we should also check that the SA with the given name exists.

Reply:
At Line 114 i am checking if the tier.Spec.SpaceRequestConfig is not nil,is that fine?

Reply:
At L488 we are calling the method to create a token for the service account,( `generateKubeConfig` function CreateTokenRequest called)... So if the service account does not exists this should return error. Do you think that is good enough

Reply:
> At L488 we are calling the method to create a token for the service account,( generateKubeConfig function CreateTokenRequest called)...

I think that might be enough, maybe it's not worth trying to get the SA resource. 👍